pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -227,15 +227,15 @@ public void accept(Notification n) {
         }
         // Ensure caches are invalidated before the operation is confirmed
         return storeDelete(path, expectedVersion)
-                .thenRun(() -> {
+                .thenRunAsync(() -> {
```

Comment:
`storeDelete` should ensure that the callback is called in executor?

Reply:
+1, is it necessary to use `thenRunAsync`?

Reply:
@Jason918  @codelipenghui Sorry for the late response. I think we should ensure, In this way, the cache can be prevented from being executed concurrently

Reply:
You cannot use the executor to execute this, this callback thread is sometimes executed in the executor thread: https://github.com/apache/pulsar/blob/master/pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java#L330

Reply:
> In this way, the cache can be prevented from being executed concurrently

I think it's better to ensure that in `storeDelete`, to reduce this extra context switch.

Reply:
> You cannot use the executor to execute this, this callback thread is sometimes executed in the executor thread: https://github.com/apache/pulsar/blob/master/pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java#L330

There seems to be this problem.

Reply:
@nodece @Jason918 @codelipenghui I resubmitted PR #13911 , please have a look, thank you.